moderntribe/square-one
# Feature/composer v2

## What does this do/fix?

- Updates the docker image, composer.json and wp-config-environment.php for composer v2 support (requires `so` v5.3.0+)
- Updates GitHub actions to use composer v2 and specify PHP version where appropriate
- Updates some plugins
- Removes some unused dependencies

## QA

If you're going to update your own project (composer v2 is quite a bit faster!) or test this, ensure you and your team does the following:

1. Everyone is updated to `so` version 5.3.0+
1. You've reviewed the commits in this PR and applied the appropriate changes to your project
1. Stop the project (`so stop`)
1. Delete everything **but auth.json** in `dev/docker/composer` to remove globally installed composer packages
1. `so start`

> The reason we need to delete things in the `dev/docker/composer` folder is because we install [Prestissimo](https://github.com/hirak/prestissimo) for composer v1 to speed up downloads, but this isn't compatible with composer v2, so projects that have been previously started with composer v1 will have that installed still.

**Note:** only the PHP 7.4 docker container has been upgraded with composer v2 in the 3.0 branch. If you need 7.2, 7.3 with composer v2, reach out to me.

## Tests

Does this have tests?

- [x] No, this doesn't need tests because it's a composer update.
